tests/testthat/test-emmeans.R

```diff
@@ -148,3 +148,29 @@ test_that("emmeans works as expected also for weighted model", {
   )
   expect_equal(result_emmeans_df, expected_emmeans_df, tolerance = 1e-3)
 })
+
+test_that("emmeans gives d.f. close to what is expected for weighted model", {
+  skip_if_not_installed("emmeans", minimum_version = "1.6")
+
+  formula <- FEV1 ~ RACE + SEX + ARMCD * AVISIT + us(AVISIT | USUBJID)
+  fev_data <- within(fev_data, AVISIT <- relevel(AVISIT, ref = "VIS4"))
+  fit <- mmrm(formula, fev_data, weights = fev_data$WEIGHT)
+  result <- expect_silent(emmeans::emmeans(fit, pairwise ~ ARMCD*AVISIT))
```

Comment:
need spaces around the `*` (sorry for some reason I cannot give suggestion in the code directly)